From 413416ebc1d16a8f5cc4cd187e8a36637508cfbe Mon Sep 17 00:00:00 2001
From: Henry Wang <henwang@amazon.com>
Date: Thu, 10 Apr 2025 20:50:50 +0000
Subject: [PATCH] Revert "Don't allow io_uring related syscalls in the
 RuntimeDefault seccomp profile."

This reverts commit a48ddf4a208b24eadea82f0eac62e236f2acf004.

[updated for containerd 2.3.7: v2.3.7 adds socket-domain tests to
seccomp_default_test.go (5ba51c0f2), so remove only TestIOUringIsNotAllowed
instead of deleting the file]
Signed-off-by: Kyle Sessions <kssessio@amazon.com>
---
 contrib/seccomp/seccomp_default.go      |  3 +++
 contrib/seccomp/seccomp_default_test.go | 29 -------------------------
 2 files changed, 3 insertions(+), 29 deletions(-)

diff --git a/contrib/seccomp/seccomp_default.go b/contrib/seccomp/seccomp_default.go
index eda0223..20d81db 100644
--- a/contrib/seccomp/seccomp_default.go
+++ b/contrib/seccomp/seccomp_default.go
@@ -297,6 +297,9 @@ func DefaultProfile(sp *specs.Spec) *specs.LinuxSeccomp {
 				"ioprio_set",
 				"io_setup",
 				"io_submit",
+				"io_uring_enter",
+				"io_uring_register",
+				"io_uring_setup",
 				"ipc",
 				"kill",
 				"landlock_add_rule",
diff --git a/contrib/seccomp/seccomp_default_test.go b/contrib/seccomp/seccomp_default_test.go
index c9c3a5c..cd56af6 100644
--- a/contrib/seccomp/seccomp_default_test.go
+++ b/contrib/seccomp/seccomp_default_test.go
@@ -11,35 +11,6 @@ import (
 	"github.com/opencontainers/runtime-spec/specs-go"
 )
 
-func TestIOUringIsNotAllowed(t *testing.T) {
-
-	disallowed := map[string]bool{
-		"io_uring_enter":    true,
-		"io_uring_register": true,
-		"io_uring_setup":    true,
-	}
-
-	got := DefaultProfile(&specs.Spec{
-		Process: &specs.Process{
-			Capabilities: &specs.LinuxCapabilities{
-				Bounding: []string{},
-			},
-		},
-	})
-
-	for _, config := range got.Syscalls {
-		if config.Action != specs.ActAllow {
-			continue
-		}
-
-		for _, name := range config.Names {
-			if disallowed[name] {
-				t.Errorf("found disallowed io_uring related syscalls")
-			}
-		}
-	}
-}
-
 func TestSocketSyscallsForDomains(t *testing.T) {
 	t.Parallel()
 
-- 
2.56.0

